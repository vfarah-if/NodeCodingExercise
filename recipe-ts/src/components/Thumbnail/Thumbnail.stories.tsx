import React, { Fragment } from 'react';
import { Story, Meta } from '@storybook/react/types-6-0';

import Thumbnail, { ThumbnailProps } from './Thumbnail';

export default {
	title: 'Component/Thumbnail',
	component: Thumbnail,
} as Meta;

const ThumbnailTemplate: Story<ThumbnailProps> = (args) => (
	<Fragment>
		<Thumbnail {...args} style={{border: '1px solid yellow'}} />
	</Fragment>
);

export const SimpleThumbnail = ThumbnailTemplate.bind({});
SimpleThumbnail.args = {
	imageSource: 'https://miro.medium.com/max/1592/0*aaeiaAX5kMyHCQZT.png',
	description: 'Jest logo',
};
