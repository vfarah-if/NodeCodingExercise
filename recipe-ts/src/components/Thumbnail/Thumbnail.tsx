
import React from 'react';
import './style/index.css';

export interface ThumbnailProps {
    imageSource: string,
    description?: string,
    style?:React.CSSProperties
}

const Thumbnail: React.FC<ThumbnailProps> = ({imageSource, description = '', ...rest}) => {
    return (
		<>
			<img
				className='thumbnail'
				src={imageSource}
				alt={description || ''}
				{...rest}
			/>
		</>
    )
}

export default Thumbnail;
