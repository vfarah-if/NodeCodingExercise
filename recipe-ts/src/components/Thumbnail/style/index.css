.thumbnail {
	border-radius: 4px;
	padding: 10px;
	width: 150px;
}

.thumbnail:hover {
	box-shadow: 0 0 2px 1px rgba(0, 140, 186, 0.5);
}