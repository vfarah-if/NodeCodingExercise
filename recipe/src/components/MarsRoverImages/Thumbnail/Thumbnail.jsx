import React from 'react';
import PropTypes from 'prop-types';

import './style/index.css';

const Thumbnail = ({ imageSource, description, ...rest }) => {
	return (
		<>
			<img
				className='thumbnail'
				src={imageSource}
				alt={description || ''}
				{...rest}
			/>
		</>
	);
};

Thumbnail.propTypes = {
	imageSource: PropTypes.string.isRequired,
	description: PropTypes.string.isRequired,
};

Thumbnail.defaultProps = {
	description: '',
};

export default Thumbnail;
