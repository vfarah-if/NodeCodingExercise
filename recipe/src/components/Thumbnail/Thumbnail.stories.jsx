import React from 'react';
import Thumbnail from './Thumbnail';

// eslint-disable-next-line import/no-anonymous-default-export
export default {
	title: 'Component/Thumbnail',
	component: Thumbnail,
};

const ThumbnailTemplate = (args) => (
	<>
		<Thumbnail {...args} style={{border: '1px solid yellow'}}  />
	</>
);

export const SimpleThumbnailStory = ThumbnailTemplate.bind({});
SimpleThumbnailStory.args = {
	imageSource: 'https://miro.medium.com/max/1592/0*aaeiaAX5kMyHCQZT.png',
	description: 'Jest logo',
};
