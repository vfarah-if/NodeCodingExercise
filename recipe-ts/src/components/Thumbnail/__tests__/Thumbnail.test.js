import React from 'react';
import { render } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import { SimpleThumbnail } from '../Thumbnail.stories';

describe('Cards', () => {
	it('should create a simple thumbnail with an alt and style defined', () => {
		const { container } = render(
			<SimpleThumbnail
				imageSource={SimpleThumbnail.args.imageSource}
                description={SimpleThumbnail.args.description}
                style={SimpleThumbnail.args.style}
			/>
		);

		expect(container).toMatchSnapshot();
	});
});
