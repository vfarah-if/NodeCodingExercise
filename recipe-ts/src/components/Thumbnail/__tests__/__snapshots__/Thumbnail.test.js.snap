// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cards should create a simple thumbnail with an alt and style defined 1`] = `
<div>
  <img
    alt="Jest logo"
    class="thumbnail"
    src="https://miro.medium.com/max/1592/0*aaeiaAX5kMyHCQZT.png"
    style="border: 1px solid yellow;"
  />
</div>
`;
